crypto/src/asset.rs

```diff
@@ -38,6 +38,26 @@ impl std::fmt::Debug for Id {
     }
 }
 
+impl std::fmt::Display for Id {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        use ark_ff::BigInteger;
+        let bytes = self.0.into_repr().to_bytes_le();
+        f.write_fmt(format_args!("asset::Id({})", hex::encode(&bytes)))
+    }
+}
+
+impl std::str::FromStr for Id {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let bytes: [u8; 32] = s.as_bytes().try_into().map_err(|_| {
+            anyhow::anyhow!("could not deserialize Asset ID: input vec is not 32 bytes")
+        })?;
+        let inner = Fq::from_bytes(bytes)?;
+        Ok(Id(inner))
+    }
+}
+
```

Comment:
non-blocking for this PR: should we use bech32 for asset ids? off the top of my head:

- pros: error-correction, prefix can identify what the data is (e.g., `passet1aedfea...`)
- cons: slightly longer encodings (it's already 32 bytes, does this matter? doubt people will be typing manually)

Reply:
i think this is a great idea

Reply:
I don't think that the slightly longer encodings matter, and having a checksum is nice